### ISSUE TITLE
@rtk-query/codegen triggers ERR_REQUIRE_ESM when using prettier with plugins

### ISSUE THREAD

**Issue by vmmitev** [reduxjs/redux-toolkit#3792]
When using the **prettier-plugin-jsdoc** (current latest version 1.1.1) in my **.prettierrc** as such:
```
  "plugins": [
    "prettier-plugin-jsdoc"
  ],
```

and running ```npx @rtk-query/codegen-openapi ./api.config.ts``` I get the following error:
```
Error [ERR_REQUIRE_ESM]: require() of ES Module <projectRoot>\node_modules\prettier-plugin-jsdoc\dist\index.js from <projectRoot>\node_modules\@rtk-query\codegen-openapi\node_modules\prettier\index.js not supported.
Instead change the require of <projectRoot>\node_modules\prettier-plugin-jsdoc\dist\index.js in <projectRoot>\node_modules\@rtk-query\codegen-openapi\node_modules\prettier\index.js to a dynamic import() which is available in all CommonJS modules.
    at require.extensions.<computed> [as .js] (<projectRoot>\node_modules\ts-node\dist\index.js:851:20)
    at <projectRoot>\node_modules\@rtk-query\codegen-openapi\node_modules\prettier\index.js:38143:10
    at Array.map (<anonymous>)
    at Object.load (<projectRoot>\node_modules\@rtk-query\codegen-openapi\node_modules\prettier\index.js:38141:128)
    at Object.load [as loadPlugins] (<projectRoot>\node_modules\@rtk-query\codegen-openapi\node_modules\prettier\index.js:16147:23)
    at <projectRoot>\node_modules\@rtk-query\codegen-openapi\node_modules\prettier\index.js:38181:24
    at Object.format (<projectRoot>\node_modules\@rtk-query\codegen-openapi\node_modules\prettier\index.js:38197:12)
    at prettify (<projectRoot>\node_modules\@rtk-query\codegen-openapi\lib\utils\prettier.js:57:21)
    at async generateEndpoints (<projectRoot>\node_modules\@rtk-query\codegen-openapi\lib\index.js:21:87)
    at async run (<projectRoot>\node_modules\@rtk-query\codegen-openapi\lib\bin\cli.js:54:13) {
  code: 'ERR_REQUIRE_ESM'
}
```

The code generator correctly uses prettier to dynamically import the plugin but it uses ```require``` and thus produces the error in runtime. There might be other configurations in the project's context that cause this error but when I remove the plugin in **.prettierrc** the command runs without error. I assume any other prettier plugins will also result in the same issue.

Is there any change to the TS Node context or prettier utils that could alleviate this issue?

**Comment by markerikson** [comment]
We probably need to apply the same package changes we're doing for all the other Redux packages in the upcoming major versions.  Just hadn't even thought to do that for the codegen yet.

**Comment by phryneas** [comment]
Could it be that you are on prettier v3? The Codegen just supports v2 for now.

**Comment by vmmitev** [comment]
Yes indeed I have prettier v3.0.3 installed in my root project. However, in the node_modules the codegen-openapi tool has the correct v2.2.1 resolved for its imports. I assume it should be using the expected scripts in runtime.

**Comment by henkkasoft** [comment]
This might be solved in 2.0.0?

**Comment by markerikson** [comment]
I'll assume it is and close this.
